### PR TITLE
add Smart Proxy plugin monitoring

### PR DESCRIPTION
**Pull request by dgoetz**
Please build for:

* [x] Nightly
* [x] 1.12
* [ ] 1.11
* [ ] 1.10

### CODE REVIEW

File: rubygem-smart_proxy_monitoring/rubygem-smart_proxy_monitoring.spec [hunk 1]
```diff
@@ -0,0 +1,98 @@
+%global gem_name smart_proxy_monitoring
+%global plugin_name monitoring
+
+%global foreman_proxy_dir %{_datarootdir}/foreman-proxy
+%global foreman_proxy_bundlerd_dir %{foreman_proxy_dir}/bundler.d
+%global foreman_proxy_settingsd_dir %{_sysconfdir}/foreman-proxy/settings.d
+
+Name: rubygem-%{gem_name}
+Version: 1.0.0
```
Comment (domcleal) [inline rubygem-smart_proxy_monitoring/rubygem-smart_proxy_monitoring.spec]: `0.0.1`, which might be the cause of the test failure

File: rubygem-smart_proxy_monitoring/rubygem-smart_proxy_monitoring.spec [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+Release: 1%{?foremandist}%{?dist}
+Summary: Monitoring smart proxy plugin
+Group: Applications/Internet
+License: GPLv3
+URL: http://github.com/filiadata/smart_proxy_monitoring
```
Comment (domcleal) [inline rubygem-smart_proxy_monitoring/rubygem-smart_proxy_monitoring.spec]: theforeman?

**Comment by dgoetz** [comment]
Sorry copied the initial test version instead of the last one.

File: rubygem-smart_proxy_monitoring/rubygem-smart_proxy_monitoring.spec [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+%global foreman_proxy_settingsd_dir %{_sysconfdir}/foreman-proxy/settings.d
+
+Name: rubygem-%{gem_name}
+Version: 0.0.1
+Release: 1%{?foremandist}%{?dist}
+Summary: Monitoring smart proxy plugin
+Group: Applications/Internet
+License: GPLv3
+URL: http://github.com/theforeman/smart_proxy_monitoring
+Source0: %{gem_name}-%{version}.gem
```
Comment (domcleal) [inline rubygem-smart_proxy_monitoring/rubygem-smart_proxy_monitoring.spec]: This should be `http://rubygems.org/downloads/%{gem_name}-%{version}.gem` for the full download location.

File: rubygem-smart_proxy_monitoring/rubygem-smart_proxy_monitoring.spec [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+# sample config
+mkdir -p %{buildroot}%{foreman_proxy_settingsd_dir}
+mv %{buildroot}%{gem_instdir}/settings.d/monitoring.yml.example \
+   %{buildroot}%{foreman_proxy_settingsd_dir}/monitoring.yml
+mv %{buildroot}%{gem_instdir}/settings.d/monitoring_icinga2.yml.example \
+   %{buildroot}%{foreman_proxy_settingsd_dir}/monitoring_icinga2.yml
+
+# additional config
+mkdir -p %{buildroot}%{foreman_proxy_dir}/monitoring
```
Comment (domcleal) [inline rubygem-smart_proxy_monitoring/rubygem-smart_proxy_monitoring.spec]: Does this do anything? I can't see how it's used.

Comment (dgoetz) [inline rubygem-smart_proxy_monitoring/rubygem-smart_proxy_monitoring.spec]: We need a location for additional configuration like the certificates and key for communication with Icinga 2. I did not want to bloat /etc/foreman/settings.d and other plugins also used a subdir of /usr/share/foreman. Would you prefer another location?

Comment (domcleal) [inline rubygem-smart_proxy_monitoring/rubygem-smart_proxy_monitoring.spec]: Yeah, I'd suggest keeping the program files separate to anything added by the user, it'll make it easier for users to back up.  /etc/foreman-proxy is probably the right location - I'd suggest creating /etc/foreman-proxy/monitoring, then if you need /usr/share/foreman-proxy/monitoring/ to exist too, add a symlink from /usr to /etc/.

File: rubygem-smart_proxy_monitoring/rubygem-smart_proxy_monitoring.spec [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+%config(noreplace) %attr(0640, root, foreman-proxy) %{foreman_proxy_settingsd_dir}/monitoring.yml
+%config(noreplace) %attr(0640, root, foreman-proxy) %{foreman_proxy_settingsd_dir}/monitoring_icinga2.yml
+%doc %{gem_instdir}/LICENSE
+%{gem_instdir}/bundler.d
+%{gem_libdir}
+%{gem_instdir}/settings.d
+%{foreman_proxy_bundlerd_dir}/%{plugin_name}.rb
+%exclude %{gem_cache}
+%{gem_spec}
+%{foreman_proxy_dir}/monitoring
```
Comment (domcleal) [inline rubygem-smart_proxy_monitoring/rubygem-smart_proxy_monitoring.spec]: add %dir to the start, as there are no files in it

**Comment by domcleal** [comment]
Thanks @dgoetz, merged and building for both releases now. It should be available within a day or so.

**Pull request merged**